src/D2L.Security.OAuth2/Validation/AccessTokens/AccessTokenValidator.cs

```diff
@@ -36,8 +36,14 @@ IPublicKeyProvider publicKeyProvider
 			string token
 		) {
 			var tokenHandler = m_tokenHandler.Value;
+			var tokenInParts = token.Split( '.' );	// header.payload.signature
 
-			if( !tokenHandler.CanReadToken( token ) ) {
+			if(
+				!tokenHandler.CanReadToken( token ) ||
+				// https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/482
+				!IsValidPart( tokenInParts[0] ) ||
+				!IsValidPart( tokenInParts[1] )
+			) {
 				throw new ValidationException( "Couldn't parse token" );

```

Comment:
Seems right. You can fix that in a patch revision IMO (base class etc.) so that people can handle exceptions more specifically if they want.